src/App.tsx

```diff
@@ -1,41 +1,93 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { Post } from './types/post';
+import { getUserPosts } from './api/posts';
+import { getUsers } from './api/users';
+import { User } from './types/user';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      const response = await getUsers();
+
+      setUsers(response);
+    };
+
+    const fetchUserPosts = async () => {
+      const response = await getUserPosts(selectedUserId);
+
+      setPosts(response);
+    };
+
+    try {
+      fetchUsers();
+    } catch (error) {
+      throw new Error(`${error}`);
+    }
+
+    try {
+      fetchUserPosts();
+    } catch (error) {
+      throw new Error(`${error}`);
+    }
+  }, [selectedUserId]);
+
+  const handleSelectedUserId = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setSelectedUserId(Number(event.target.value));
+  };
+
+  const handleSelectedPostId = useCallback((postId: number) => {
+    setSelectedPostId(postId);
+  }, [selectedPostId]);
+
+  return (
+    <div className="App">
+      <header className="App__header">
+        <label>
+          Select a user: &nbsp;
```

Comment:
why do you need this?
```suggestion
          Select a user:
```